model/Bom.js

```diff
@@ -38,11 +41,46 @@ class Bom extends CycloneDXObject {
     if (pkg) {
       this._metadata = this.createMetadata(pkg, componentType)
       this._components = this.listComponents(pkg, lockfile)
+      this._dependencies = this.listDependencies(pkg)
     } else {
       this._components = []
+      this._dependencies = []
     }
   }
 
+  listDependencies (pkg) {
+    const list = []
+    this.createDependency(pkg, list)
+    return list
+  }
+
+  createDependency (pkg, list) {
+    // read-installed with default options marks devDependencies as extraneous
+    // if a package is marked as extraneous, do not include it as a dependency
+    if (pkg.extraneous) return
+    const rootBomRef = this.createBomRef(pkg)
+    const deplist = []
+    if (pkg._dependencies && Object.keys(pkg._dependencies).length !== 0) {
+      Object.keys(pkg._dependencies)
+        .map(x => pkg.dependencies[x])
+        .filter(x => x !== undefined) // remove cycles
+        .map(x => deplist.push(new Dependency(this.createBomRef(x), this.createDependency(x, list))))
+      list.push(new Dependency(rootBomRef, deplist))
+    }
+    return deplist
+  }
+
+  createBomRef (pkg) {
```

Comment:
:x: rename to "createPackageURL"

a bom-ref is defined as a string. 
it might look like a PURL, but it must not. it even could be an URN

Reply:
Makes sense, I will change it.

Reply:
thanks for fixing